### PR TITLE
feat(ui-core): implement drilldown

### PR DESCRIPTION
**Pull request by EvilDrW**
Add hierarchical traversability to pie charts and bar charts.

### SUMMARY
This is an attempt at bringing https://github.com/apache/superset/pull/14688 and https://github.com/apache-superset/superset-ui/pull/1120 up-to-date with the latest Superset mono repo and extending the feature to work both with echarts/pie and nvd3/distbar.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
See video on https://github.com/apache/superset/pull/14688

### TESTING INSTRUCTIONS
Create either a bar chart or pie chart.  In the chart editor, select "ENABLE DRILL DOWN."  Add multiple columns to the "GROUP BY" field that have a natural hierarchy to them and ensure the order of columns listed reflects the intended hierarchy.  Add an appropriate metric (usually COUNT(id) or similar).  Observe that only the first column in the hierarchy is reflected in the rendered chart.  Observe that clicking one field in the chart adds the selected field to the applied filters and that the second column in the "GROUP BY" hierarchy is now displayed.  Observe that selecting a field when the chart is at its deepest level in the hierarchy clears all selected filters and returns the chart to the original state.  Observe that saving the chart and adding it to a dashboard results in identical functionality as described above.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [x] Required feature flags: DASHBOARD_DRILL_DOWN
- [x] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [x] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by Vygo** [comment]
Super excited about this feature, any chance we can it in the next release?

**Comment by villebro** [comment]
As excited as I am about this feature, I think we need to do SIP for this to ensure that we don't end up painting ourselves into a corner by making changes that will become difficult to maintain going forward. For context, the original POC was met with some concern in offline discussions, primarily around what additional changes we would need to introduce to make drill down/drill through functionality in a future proof way without adding technical debt. @EvilDrW would you be open to spearheading a SIP to kickstart the discussion for adding drilldown functionality to Superset? This way we could have a transparent discussion where all parties could be heard, and could also make sure this feature could be rolled out incrementally with a clear implementation plan.

**Comment by EvilDrW** [comment]
https://github.com/apache/superset/issues/17927

File: superset-frontend/plugins/legacy-preset-chart-nvd3/src/DistBar/controlPanel.ts [hunk 1]
```diff
@@ -75,6 +81,23 @@ const config: ControlPanelConfig = {
             },
           },
         ],
+        isFeatureEnabled(FeatureFlag.DASHBOARD_DRILL_DOWN)
```
Comment (rajatency) [inline superset-frontend/plugins/legacy-preset-chart-nvd3/src/DistBar/controlPanel.ts:84]: Weird heirachy of [] and {}

**Comment by rajatency** [comment]
@EvilDrW  I am trying to implement this for the purpose of POC. How can I add this to other types of charts as well, apart from pie chart and bar chart? Also, I see I can add multiple group-by to pie chart and use this drill down feature, but how will this be visible in bar charts?

**Comment by alfgai** [comment]
hi everyone!
Is this option already available? i've installed Apache using docker a week ago and I don't have "enable drill-down" checkbox.

**Comment by ajaykumarparashar11** [comment]
@EvilDrW is this code available in the latest master branch? if yes then how can i enable this, if not then can you please guide me to implement this in the current latest master branch, so that i can use this feature for now?

**Comment by namalashivacharan123** [comment]
> @EvilDrW is this code available in the latest master branch? if yes then how can i enable this, if not then can you please guide me to implement this in the current latest master branch, so that i can use this feature for now?



> @EvilDrW is this code available in the latest master branch? if yes then how can i enable this, if not then can you please guide me to implement this in the current latest master branch, so that i can use this feature for now?

Did you find any answer..?

**Comment by sundar68** [comment]
Does anyone tried this drilldown on table chart. If yes, Can you please explain me.

**Comment by rusackas** [comment]
Drill By (flexible exploration) is Superset's preferred solution over Drill Down (strict hierarchy). It's available in Preset and OSS Superset: https://preset.io/blog/drill-by/

**Comment by sundar68** [comment]
@EvilDrW 
Can you please help me on how to update dashboard native filters from inside the plugin.

**Pull request closed without merging**